moirae/estimators/online/filters/conversions.py

```diff
@@ -0,0 +1,191 @@
+        """
+        raise NotImplementedError('Implement in child class!')
+
+
+class IdentityConversionOperator(ConversionOperator):
+    """
+    Class that implements simple identity operation, that is, it does not change the inputs.
+    """
+    def transform_samples(self, samples: Optional[np.ndarray]) -> np.ndarray:
+        if samples is None:
+            return
+        return samples.copy()
+
+    def transform_covariance(self, covariance: Optional[np.ndarray]) -> np.ndarray:
+        if covariance is None:
+            return
+        return covariance.copy()
+
+    def inverse_transform_samples(self, transformed_samples: Optional[np.ndarray]) -> np.ndarray:
+        if transformed_samples is None:
+            return
+        return transformed_samples.copy()
+
+    def inverse_transform_covariance(self, transformed_covariance: Optional[np.ndarray]) -> np.ndarray:
+        if transformed_covariance is None:
+            return
+        return transformed_covariance.copy()
+
+
+class LinearConversionOperator(ConversionOperator, validate_assignment=True):
+    """
+    Class that implements a linear function as a transformation (strictly speaking, this is not a linear transformation,
+    but just a linear function).
+
+    Given an array of multiplicative factors ``multi`` and additive factors ``bias``, this function transforms points
+    ``x`` to ``y`` following ``y = (multi * x) + bias``
+
+    Args:
+        multiplicative_array: np.ndarray corresponding to multiplicative factors in the linear function
+        additive_array: np.ndarray corresponding to additive (bias) factors in the linear function
+    """
+    multiplicative_array: np.ndarray = Field(description='Multiplicative factors of linear function',
+                                             default=np.array(1.))
+    additive_array: Optional[np.ndarray] = Field(description='Additive (bias) factors of linear function',
+                                                 default=np.array([0.]))
+
+    @field_validator('additive_array', mode='after')
+    @classmethod
+    def additive_1d(cls, bias: np.ndarray) -> np.ndarray:
+        """
+        Ensures the additive array is a 1D vector, and not a matrix
+        """
+        shape = bias.shape
+        if len(shape) > 1:
+            raise ValueError(f'Additive factor must be a vector or scalar, but, instead, is has shape {shape}!')
+        return bias.flatten()
+
+    @field_validator('multiplicative_array', mode='after')
+    @classmethod
+    def multiplicative_2d(cls, multi: np.ndarray) -> np.ndarray:
+        """
+        Ensures the multiplicative array is stored as a 2D array (or as a shapeless object)
+        """
+        shape = multi.shape
+        if len(shape) > 2:
+            raise ValueError(f'Additive factor must be at most 2D, but, instead, is has shape {shape}!')
+        if len(shape) == 1:
+            return np.diag(multi)
+        return multi
+
+    @computed_field
```

Comment:
Could you change this from "computed_field," which adds the value to the serialized version of the class to "cached_property?" We don't need the additional functionality which comes with storing the values in the JSON form of this class.

Reply:
Sure, I had added it here because it caches it, but there are other ways of doing the same thing without the pydantic overhead. Taking care of it.